File: app/src/main/java/com/dhu/dhu_timetable/ui/navitem/notice/NoticeViewModel.java
package com.dhu.dhu_timetable.ui.navitem.notice;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.dhu.dhu_timetable.model.NoticeModel;
import com.dhu.dhu_timetable.repo.NoticeRepo;
import java.util.List;

public class NoticeViewModel extends ViewModel {
    private MutableLiveData<List<NoticeModel>> noticeModels;
    private List<NoticeModel> noticeModelList;
    private NoticeRepo noticeRepo;

    public NoticeViewModel() {
        if (noticeRepo != null) {
            return;
        }
        noticeRepo = NoticeRepo.getInstance();
    }

    public LiveData<List<NoticeModel>> getNoticeData() {
        return noticeRepo.getNoticeData();
    }

    public void setNoticeData() {
        noticeRepo.setNoticeData();
    }
}

File: app/src/main/java/com/dhu/dhu_timetable/ui/navitem/notice/NoticeViewModel.kt
package com.dhu.dhu_timetable.ui.navitem.notice

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.dhu.dhu_timetable.model.NoticeModel
import com.dhu.dhu_timetable.repo.NoticeRepo
import com.dhu.dhu_timetable.repo.NoticeRepo.Companion.getInstance
import java.lang.Exception


class NoticeViewModel : ViewModel() {
    private lateinit var noticeRepo: NoticeRepo

    val noticeData: LiveData<List<NoticeModel>>
        get() = noticeRepo.getNoticeData()

    fun setNoticeData() {
        noticeRepo.setNoticeData()
    }

//    fun getNoticeData(): LiveData<List<NoticeModel>> = noticeData

    init {
        try {
            noticeRepo = getInstance()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}